### PR TITLE
Fix #639: add testing of CreateCollection with mixed-case name, fix quoting for FindCollections

### LINKED ISSUE #639: Verify handling of Collection with capital letter(s)
(user reported)

Use of mixed-case letters is allowed for Collections (which map to Cassandra Tables). But it seems that when reporting the name, it is sometimes surrounded by double-quotes, likely due to internal handling for CQL.
This should not be exposed to JSON API users: if and when quoting is needed for CQL that should be handled transparently.

Two places where this may be exposed would be:

1. Response when creating Collection (`CreateCollection` command)
2. `FindCollections` command

and so need to add ITs and fix handling if necessary.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds tests for #639 to check that mixed-case Collection names are supported, and also fix one issue found: that of unwanted quoting of Collection name in mixed-case case.

**Which issue(s) this PR fixes**:
Fixes #639

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java [hunk 1]
```diff
@@ -59,7 +59,8 @@ public static SimilarityFunction fromString(String similarityFunction) {
 
   public static CollectionSettings getCollectionSettings(
       TableMetadata table, ObjectMapper objectMapper) {
-    String collectionName = table.getName().asCql(true);
+    // [jsonapi#639]: get internal name to avoid quoting of case-sensitive names
+    String collectionName = table.getName().asInternal();
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java:63]: This is the actual fix: prevents name from being surrounded by double-quotes (needed for CQL usage, but not in JSON)

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/AbstractCollectionIntegrationTestBase.java [hunk 2]
```diff
@@ -26,21 +26,23 @@ public abstract class AbstractCollectionIntegrationTestBase
...
-          "createCollection": {
-            "name": "%s"
-          }
-        }
-        """
-            .formatted(collectionName);
+    createCollection(this.collectionName);
+  }
 
+  protected void createCollection(String collectionToCreate) {
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/AbstractCollectionIntegrationTestBase.java:32]: Minor refactoring to make it easy to create custom collections from ITs.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindCollectionsIntegrationTest.java [hunk 3]
```diff
@@ -79,7 +74,7 @@ public void happyPath() {
     }
 
     @Test
-    @Order(1)
+    @Order(2)
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindCollectionsIntegrationTest.java:77]: Had same order, assuming accidentally, fixed.

**Review by Hazel-Datastax — APPROVED** [review]
LGTM

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:

**Pull request merged**